src/plugins/intel_cpu/src/infer_request.cpp

```diff
@@ -342,600 +354,350 @@ void InferRequestBase::changeDefaultPtr() {
-            externalPtr.erase(name);
+            external_ptr[name] = tensor;
+        } else if (external_ptr.find(name) != external_ptr.end()) {
+            external_ptr.erase(name);
         }
-        _inputs[name] = data;
-        _batched_inputs.erase(name);
     } else {
-        if (compoundBlobPassed) {
-            IE_THROW(NotImplemented) << "Can't set compound blob: supported only for input pre-processing";
-        }
-        const auto netOutPrc = InferenceEngine::details::convertPrecision(outputNodeItr->second->get_input_element_type(0));
-        if (netOutPrc != blobDesc.getPrecision()) {
-            IE_THROW(ParameterMismatch) << "Failed to set input blob with precision: "
-                               << blobDesc.getPrecision() << ", if CNNNetwork output blob precision is: " << netOutPrc;
+        const auto netOutPrc = port.get_element_type();
+        if (netOutPrc != tensor->get_element_type()) {
+            IE_THROW(ParameterMismatch) << "Failed to set output tensor with precision: " << tensor->get_element_type()
+                                        << ", if model output tensor precision is: " << netOutPrc;
         }
 
-        const auto shape = outputNodeItr->second->get_input_partial_shape(0);
+        const auto& shape = port.get_partial_shape();
         const bool isDynamic = shape.is_dynamic();
 
-        if (!shape.compatible(ov::PartialShape(data->getTensorDesc().getDims()))) {
-            IE_THROW() << "Can't set output blob with name: " << name
-                       << ", because model output (shape=" << shape
-                       << ") and blob (shape=" << vec2str(data->getTensorDesc().getDims()) << ") are incompatible";
+        if (!shape.compatible(ov::PartialShape(tensor->get_shape()))) {
+            OPENVINO_THROW("The tensor size is not equal to model, can't set output tensor with name: ",
+                           name,
+                           ", because model output (shape=",
+                           shape,
+                           ") and blob (shape=",
+                           vec2str(tensor->get_shape()),
+                           ") are incompatible");
         }
 
-        if (!isDynamic && ngraph::shape_size(shape.to_shape()) != data->size()) {
-            IE_THROW() << "Can't set output blob with name: " << name << ", because model output size = " << ngraph::shape_size(shape.to_shape())
-                       << " and blob size = " << data->size() << " are different.";
+        if (!isDynamic && ov::shape_size(shape.to_shape()) != tensor->get_size()) {
+            OPENVINO_THROW("Can't set output tensor with name: ",
+                           name,
+                           ", because model output size = ",
+                           ov::shape_size(shape.to_shape()),
+                           " and blob size = ",
+                           tensor->get_size(),
+                           " are different.");
```

Comment:
```suggestion
            OPENVINO_THROW("Can't set the output tensor with name: ",
                           name,
                           ", because the model output tensor size = ",
                           ov::shape_size(shape.to_shape()),
                           " and the current tensor size = ",
                           tensor->get_size(),
                           " are different.");
```